### PR TITLE
Add metadata state for HCP Packer Registry artifact metadata

### PR DESCRIPTION
**Pull request by nywilken**
Needed by https://github.com/hashicorp/packer/pull/11154

### CODE REVIEW

File: builder/common/artifact.go [hunk 1]
```diff
@@ -66,6 +71,8 @@ func (a *Artifact) State(name string) interface{} {
 	switch name {
 	case "atlas.artifact.metadata":
 		return a.stateAtlasMetadata()
+	case "par.artifact.metadata":
+		return a.statePARMetadata()
```
Comment (sylviamoss) [inline builder/common/artifact.go:75]:
```suggestion
		return a.stateHCPPackerRegistryMetadata()
```

File: builder/common/artifact.go [hunk 2]
```diff
@@ -119,3 +126,18 @@ func (a *Artifact) stateAtlasMetadata() interface{} {
 
 	return metadata
 }
+
+func (a *Artifact) statePARMetadata() interface{} {
```
Comment (sylviamoss) [inline builder/common/artifact.go:130]: Updating the name like Megan did for the Packer code
```suggestion
// stateHCPPackerRegistryMetadata will write the metadata as an hcpRegistryImage for each of the AMIs 
// present in this artifact. 
func (a *Artifact) stateHCPPackerRegistryMetadata() interface{} {
```

File: builder/common/artifact.go [hunk 3]
```diff
@@ -28,6 +28,11 @@ type Artifact struct {
 	Session *session.Session
 }
 
+type registryImage struct {
```
Comment (sylviamoss) [inline builder/common/artifact.go:31]:
```suggestion
// hcpPackerRegistryImage provides the necessary metadata information that Packer will push to HCP Packer Registry
type hcpPackerRegistryImage struct {
```

File: builder/common/artifact.go [hunk 4]
```diff
@@ -119,3 +126,18 @@ func (a *Artifact) stateAtlasMetadata() interface{} {
 
 	return metadata
 }
+
+func (a *Artifact) statePARMetadata() interface{} {
+	metadata := make([]registryImage, 0, len(a.Amis))
+
+	for region, imageId := range a.Amis {
+		metadata = append(metadata, registryImage{
```
Comment (sylviamoss) [inline builder/common/artifact.go:134]:
```suggestion
		metadata = append(metadata, hcpPackerRegistryImage{
```

File: builder/common/artifact_test.go [hunk 5]
```diff
@@ -88,3 +88,28 @@ func TestArtifactState(t *testing.T) {
...
+
+func TestArtifactState_parMetadata(t *testing.T) {
+	a := &Artifact{
+		Amis: map[string]string{
+			"east": "foo", "west": "bar",
+		},
+	}
+
+	actual := a.State("par.artifact.metadata")
+	expected := []registryImage{
```
Comment (sylviamoss) [inline builder/common/artifact_test.go:100]:
```suggestion
	expected := []hcpPackerRegistryImage{
```

File: builder/common/artifact_test.go [hunk 6]
```diff
@@ -88,3 +88,28 @@ func TestArtifactState(t *testing.T) {
 		t.Fatalf("Bad: State should be nil for nil StateData")
 	}
 }
+
+func TestArtifactState_parMetadata(t *testing.T) {
```
Comment (sylviamoss) [inline builder/common/artifact_test.go:92]:
```suggestion
func TestArtifactState_hcpPackerRegistryMetadata(t *testing.T) {
```

**Review by sylviamoss — APPROVED** [review]
This looks good! 👍🏼  🎉  I added some comments and renamed PAR to HCP Packer Registry. 
Feel free to change the comments with a better text :D

File: builder/common/artifact.go [hunk 7]
```diff
@@ -119,3 +129,20 @@ func (a *Artifact) stateAtlasMetadata() interface{} {
 
 	return metadata
 }
+
+// stateHCPPackerRegistryMetadata will write the metadata as an hcpRegistryImage for each of the AMIs 
+// present in this artifact. 
+func (a *Artifact) stateHCPPackerRegistryMetadata() interface{} {
+	metadata := make([]registryImage, 0, len(a.Amis))
```
Comment (nywilken) [inline builder/common/artifact.go:136]:
```suggestion
	metadata := make([] hcpPackerRegistryImage, 0, len(a.Amis))
```

**Comment by nywilken** [comment]
> This looks good! 👍🏼 🎉 I added some comments and renamed PAR to HCP Packer Registry.
> Feel free to change the comments with a better text :D

Awesome, thanks for the review. The changes look great. I committed and will merge once all is green.

File: builder/common/artifact.go [hunk 8]
```diff
@@ -119,3 +129,20 @@ func (a *Artifact) stateAtlasMetadata() interface{} {
 
 	return metadata
 }
+
+// stateHCPPackerRegistryMetadata will write the metadata as an hcpRegistryImage for each of the AMIs 
+// present in this artifact. 
+func (a *Artifact) stateHCPPackerRegistryMetadata() interface{} {
+	metadata := make([] hcpPackerRegistryImage, 0, len(a.Amis))
```
Comment (nywilken) [inline builder/common/artifact.go:136]:
```suggestion
	metadata := make([]hcpPackerRegistryImage, 0, len(a.Amis))
```

**Pull request merged**